### ISSUE TITLE
A11y keyboard focus control in the Progress Spinner Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/components#3708]
#### Bug, feature request, or proposal:
Bug: Missing element text label.

#### What is the expected behavior?
All page elements are expected to have a meaningful text label.

#### What is the current behavior?
Currently the Slider element has no associated text label.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Create a good screen reader user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer 11, and Google Chrome, on Windows10.

#### Is there anything else we should know?
Screen reader user experience follows:

1. Press Tab key to the "Configurable progress-bar" demo component in the Tab Panel.
JAWS says "Primary radio button checked 1 of 3", and the braille display shows 
"<x> Primary rbtn 1 of 3".
Note, this demo uses two Radio Button groups "Color:" and "Mode:", in which the Arrow keys rotate between the grouped Buttons, as expected.
- Selecting a color does not invoke an automatic announcement of the "Result" change in the Progress Bar.
- Selecting a Mode does not invoke an automatic announcement of the "Result" change in the Progress Bar.

2. Press the Tab key to move focus to the md-slider control.
JAWS says "Left right slider 50 to increase or decrease use the arrow keys", and the braille display shows "lrtrk 50".
Note, the text label "Progress:" is not announced or displayed.

When "Determinate" is selected, focus moves to the md-slider control, and when "InDeterminate" is selected the focus moves to the page footer (which is assumed to be the expected behaviour but Indeterminate should invoke a message). This may be a feature of the demo and not a Module defect.
Note, the Slider control behaves as expected when using Arrows, Home, End, PgUP, and PgDn keys.

3. Press Tab key.
The JAWS focus skips past the "Progress bar" under the "Result" heading, and goes to the page Footer.
Although the Progress Bar value is updated, it is not announced or gets focus. This may be a feature of the demo and not a Progress Spinner defect.
